app/controllers/admin/genres_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Admin::GenresController < Admin::BaseController
+
+  def create
+    @genre = Genre.new(genre_params)
+    if @genre.save
+      flash[:success] = "You have created a #{@genre.name} genre."
+      redirect_to genres_path
+    else
+      flash[:error] = "Your genre was not successfully created."
+      redirect_to genres_path
```

Comment:
since both the if and else are doing a redirect, you could move the redirect out of the if/else block like:
```ruby
if @genre.save
  flash[:success] = '...'
else
  flash[:error] = '...'
end
redirect_to genres_path
```

Reply:
the downside to doing a redirect here instead of not namespacing this under /admin/ is that the form won't repopulate with what you have in `@genre` -- if this was a long form, users would be kinda ticked off about it.